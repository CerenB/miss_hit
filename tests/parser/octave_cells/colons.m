% Extracted from the Octave tests parser.tst
function colons()
    z = cell (1,2,3,0,5);
    assert ({1, z{:}, 2} == {1, 2});
    assert ({1; z{:}; 2} == {1; 2});
    assert ({1 2; z{:}; 3 4} == {1, 2; 3 4});
    assert ({1 2; 5 z{:} 6; 3 4} == {1, 2; 5 6; 3 4});
end
