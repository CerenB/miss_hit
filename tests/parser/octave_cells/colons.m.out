----------------------------------------------------------------------
--  Parse tree for colons.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <colons>
    Body: Sequence_Of_Statements
      Statements: Simple_Assignment_Statement
        LHS: Identifier <z>
        RHS: Reference
          Name: Identifier <cell>
          Arguments: Number_Literal <1>
          Arguments: Number_Literal <2>
          Arguments: Number_Literal <3>
          Arguments: Number_Literal <0>
          Arguments: Number_Literal <5>
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Cell_Expression
              Rows: Row
                Items: Number_Literal <1>
                Items: Cell_Reference
                  Name: Identifier <z>
                  Arguments: Reshape
                Items: Number_Literal <2>
            RHS: Cell_Expression
              Rows: Row
                Items: Number_Literal <1>
                Items: Number_Literal <2>
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Cell_Expression
              Rows: Row
                Items: Number_Literal <1>
              Rows: Row
                Items: Cell_Reference
                  Name: Identifier <z>
                  Arguments: Reshape
              Rows: Row
                Items: Number_Literal <2>
            RHS: Cell_Expression
              Rows: Row
                Items: Number_Literal <1>
              Rows: Row
                Items: Number_Literal <2>
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Cell_Expression
              Rows: Row
                Items: Number_Literal <1>
                Items: Number_Literal <2>
              Rows: Row
                Items: Cell_Reference
                  Name: Identifier <z>
                  Arguments: Reshape
              Rows: Row
                Items: Number_Literal <3>
                Items: Number_Literal <4>
            RHS: Cell_Expression
              Rows: Row
                Items: Number_Literal <1>
                Items: Number_Literal <2>
              Rows: Row
                Items: Number_Literal <3>
                Items: Number_Literal <4>
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Cell_Expression
              Rows: Row
                Items: Number_Literal <1>
                Items: Number_Literal <2>
              Rows: Row
                Items: Number_Literal <5>
                Items: Cell_Reference
                  Name: Identifier <z>
                  Arguments: Reshape
                Items: Number_Literal <6>
              Rows: Row
                Items: Number_Literal <3>
                Items: Number_Literal <4>
            RHS: Cell_Expression
              Rows: Row
                Items: Number_Literal <1>
                Items: Number_Literal <2>
              Rows: Row
                Items: Number_Literal <5>
                Items: Number_Literal <6>
              Rows: Row
                Items: Number_Literal <3>
                Items: Number_Literal <4>
----------------------------------------------------------------------
In colons.m, line 4
|     assert ({1, z{:}, 2} == {1, 2});
|                  ^ info: this token is not linked to the ast
In colons.m, line 4
|     assert ({1, z{:}, 2} == {1, 2});
|                    ^ info: this token is not linked to the ast
In colons.m, line 5
|     assert ({1; z{:}; 2} == {1; 2});
|                  ^ info: this token is not linked to the ast
In colons.m, line 5
|     assert ({1; z{:}; 2} == {1; 2});
|                    ^ info: this token is not linked to the ast
In colons.m, line 6
|     assert ({1 2; z{:}; 3 4} == {1, 2; 3 4});
|                    ^ info: this token is not linked to the ast
In colons.m, line 6
|     assert ({1 2; z{:}; 3 4} == {1, 2; 3 4});
|                      ^ info: this token is not linked to the ast
In colons.m, line 7
|     assert ({1 2; 5 z{:} 6; 3 4} == {1, 2; 5 6; 3 4});
|                      ^ info: this token is not linked to the ast
In colons.m, line 7
|     assert ({1 2; 5 z{:} 6; 3 4} == {1, 2; 5 6; 3 4});
|                        ^ info: this token is not linked to the ast
In colons.m, line 8
| end
| ^^^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
